libV2/cache/backend/redis.js

```diff
@@ -43,26 +44,19 @@ class RedisCache {
     }
 
     async getKey(key) {
-        try {
-            return this._redis.get(key);
-        } catch (error) {
-            moduleLogger
-                .with({ method: 'getKey' })
-                .error('error fetching key from redis', { key });
-            throw error;
-        }
+        return moduleLogger
+            .with({ method: 'getKey' })
+            .logAsyncError(() => this._redis.get(key),
+                'error fetching key from redis', { key });
     }
 
     async setKey(key, value) {
-        try {
-            const res = await this._redis.set(key, value);
-            return res === 'OK';
-        } catch (error) {
-            moduleLogger
-                .with({ method: 'setKey' })
-                .error('error setting key in redis', { key });
-            throw error;
-        }
+        return moduleLogger
+            .with({ method: 'setKey' })
```

Comment:
because you need to create a separate anonymous function we also can't rely on the name of this function from `func.name`... all kinda lame. We could also have automated logging the params from `...params` in the decorator but just seems like a burden without it.